File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/bpm/notifications/ManuallyChosenEntitiesTransitionNotificationModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.bpm.notifications;

import java.util.List;

import ro.cloudSoft.cloudDoc.presentation.client.shared.model.OrganizationEntityModel;

public class ManuallyChosenEntitiesTransitionNotificationModel extends TransitionNotificationModel {

	private static final long serialVersionUID = 1L;
	
	public static final String PROPERTY_MANUALLY_CHOSEN_ENTITIES = "manuallyChosenEntities";
	
	public ManuallyChosenEntitiesTransitionNotificationModel() {
		super(TransitionNotificationModelType.MANUALLY_CHOSEN_ENTITIES);
	}

	public List<OrganizationEntityModel> getManuallyChosenEntities() {
		return get(PROPERTY_MANUALLY_CHOSEN_ENTITIES);
	}
	
	public void setManuallyChosenEntities(List<OrganizationEntityModel> manuallyChosenEntities) {
		set(PROPERTY_MANUALLY_CHOSEN_ENTITIES, manuallyChosenEntities);
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/bpm/notifications/manually-chosen-entities-transition-notification.model.ts
import { TransitionNotificationModelType } from "./transition-notification-model-type";
import { TransitionNotificationModel } from "./transition-notification.model";
import { OrganizationEntityModel } from "../../organization-entity.model";
import { JsonProperty, JsonObject } from "json2typescript";

@JsonObject
export class ManuallyChosenEntitiesTransitionNotificationModel extends TransitionNotificationModel {

	public constructor() {
		super(TransitionNotificationModelType.MANUALLY_CHOSEN_ENTITIES);
	}

	@JsonProperty("manuallyChosenEntities", [OrganizationEntityModel])
	public manuallyChosenEntities: OrganizationEntityModel[];
}